x-pack/plugins/observability_ai_assistant/public/hooks/use_json_editor_model.ts

```diff
@@ -28,13 +25,20 @@ export const useJsonEditorModel = ({
 
   const [initialJsonValue, setInitialJsonValue] = useState<string | undefined>(initialJson);
 
+  const SCHEMA_URI = `http://elastic.co/${functionName}.json`;
+  const [modelUri, setModelUri] = useState<monaco.Uri | undefined>();
```

Comment:
Can you set a default value here?

Reply:
Maybe a dumb question (I'm rushing a little bit this morning), but why do we need state here?

Reply:
The model changes when the `functionName` changes. The model is the result of a `Uri.parse` call. I could just do this `Uri.parse` on every render, but this is a precaution against excessive calls to that function.

Reply:
useMemo()?

Reply:
> Can you set a default value here?

Sure. The Function editor and this hook will only be rendered if the user has selected a `functionName`. But agreed, it is safer.